File: design-patterns/strategy/ruby/plain_text_formatter.rb
class PlainTextFormatter
  def output_report(context)
    p "***** #{context.title} *****"
    context.text.each do |line|
      p line
    end
  end
end

#PLAIN_TEXT_FORMATTER = lambda do |context|
#    p "***** #{context.title} *****"
#    context.text.each do |line|
#      p line
#    end
#end

File: design-patterns/cs/Patrones/Patrones/Strategy/PlainTextFormatter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Patrones.Strategy
{
    public class PlainTextFormatter : IFormatStrategy
    {
        public string OutputReport(Report context)
        {
            string aux = string.Empty;
            aux += "*****" + context.Title + "*****" + Environment.NewLine + Environment.NewLine;

            foreach (var item in context.Body)
            {
                aux += item + Environment.NewLine;
            }

            return aux;
        }
    }
}
